### PR TITLE
resort section edtr component

### PR DESCRIPTION
**Pull request by CeEv**
https://ticketsystem.schul-cloud.org/browse/SC-2840

### CODE REVIEW

File: src/workspace/Lesson/Section/Editor.jsx [hunk 1]
```diff
@@ -43,80 +45,45 @@ export const EditorWrapper = styled.div`
...
-			hoverColor: "#B6B6B6",
-		},
-	},
+const getInitialDocValue = (section) => {
+	const { docValue } = section;
+	// if error try catch
+	return (docValue && Object.keys(docValue).length) ? docValue : { plugin: "rows" }
 }
 
+const canWrite = (section, props) => {
```
Comment (CordlessWool) [inline src/workspace/Lesson/Section/Editor.jsx]: why props and not editing directly?

File: src/workspace/Lesson/Section/Editor.jsx [hunk 2]
```diff
@@ -43,80 +45,45 @@ export const EditorWrapper = styled.div`
...
+	// if section not exist or has no key, setError
+	return section.scopePermission === 'write' && props.editing
+}
 
 export const Editor = (props) => {
```
Comment (CordlessWool) [inline src/workspace/Lesson/Section/Editor.jsx]: must not part of this pr but mabey if we start to refector we could write here the vars itself with {}? So everybody can see witch props are used or could be set

File: src/workspace/Lesson/Section/Editor.jsx [hunk 3]
```diff
@@ -126,13 +93,12 @@ export default Editor
...
 function PlainEditorContainerInner(props) {
 	const dispatch = useScopedDispatch()
-	const hasPendingChanges = useScopedSelector(hasPendingChangesSelector())
-	const [editable, setEditable] = React.useState(
-		props.editable === undefined ? true : props.editable
-	)
+	//	const hasPendingChanges = useScopedSelector(hasPendingChangesSelector())
+	//	const [editable, setEditable] = React.useState(
+	//		props.editable === undefined ? true : props.editable
+	//	)
```
Comment (CordlessWool) [inline src/workspace/Lesson/Section/Editor.jsx]: could be removed?

File: src/workspace/Lesson/Section/index.jsx [hunk 4]
```diff
@@ -93,14 +140,10 @@ const Section = ({ store, dispatch }) => {
...
-					key={section._id}
-					docValue={section.docValue}
-					id={section._id}
-					index={index}
+					key={store.view.activeSectionId}
+					section={section}
 					editing={store.view.editing}
-					dispatchChange={docValue => {
-						dispatch(updateSectionDocValue(section._id, docValue))
-					}}
```
Comment (CordlessWool) [inline src/workspace/Lesson/Section/index.jsx]: isn't it needed for sync? Did you testet the sync between to users?

**Comment by CordlessWool** [comment]
shouldn't be the Editor.jsx located in components?

File: src/workspace/Lesson/Section/index.jsx [hunk 5]
```diff
@@ -61,22 +61,69 @@ const Warning = styled(Text)`
...
+
+		setTimeoutState(setTimeout(() => {
+			setTimeoutState(null)
+			dispatch(updateSectionDocValue(store.view.activeSectionId, getDocument()))
+		}, 250))
+	}
+	const [section, index] = getSectionData(store, store.view.activeSectionId, setError)
+
+	if (!store.view.activeSectionId) {
+		setError(new Error({message: 'Kein Abschnitt ausgewählt...'}))
```
Comment (CordlessWool) [inline src/workspace/Lesson/Section/index.jsx]: should it also trigger sentry and mabey an error to the store?

File: src/workspace/Lesson/Section/Editor.jsx (hunk 3 above)
Comment (CeEv) [reply src/workspace/Lesson/Section/Editor.jsx]: I do not know is you code, i can remove it.

File: src/workspace/Lesson/Section/index.jsx [hunk 6]
```diff
@@ -61,22 +61,69 @@ const Warning = styled(Text)`
...
+
+const getErrorView = (error) => {
+	return (
+		<Container>
+			<Flex>
+				<Text center>{error.message || error}</Text>
+			</Flex>
+		</Container>
+	)
+}
```
[the remark this answers is not in the document]
Comment (CeEv) [reply src/workspace/Lesson/Section/index.jsx]: I disable this code, at the moment it do not work well

File: src/workspace/Lesson/Section/index.jsx (hunk 5 above)
Comment (CeEv) [reply src/workspace/Lesson/Section/index.jsx]: I think sentry at this position is not necessary. In the case that a real error is throw yes. Maybe we can change the error helper store and use dispatch for real errors and all the other stuff is "do anything then it work, or wait to load data".
